src/app/core/common-components/entity-fields-menu/entity-fields-menu.component.spec.ts

```diff
@@ -29,4 +30,25 @@ describe("EntityFieldsMenuComponent", () => {
   it("should create", () => {
     expect(component).toBeTruthy();
   });
+
+  it("should let a custom field override a default field", () => {
+    component.availableFields = [
+      "name",
+      {
+        id: "name",
+        label: "Custom Name",
+      },
+      {
+        id: "age",
+        label: "Age",
+      },
+    ];
+    expect(component._availableFields.length).toBe(2);
+
+    const name = component._availableFields.find((f) => f.id === "name")!;
```

Comment:
what does the `!` at the end of the statement do?

Reply:
Previously i was using the testEntity and trying to find some fields, and that time, sometimes coming back as undefined. So I used the non-null assertion (!) to avoid errors. 
But now, using/defining a custom availableFields array, I forgot to remove those. Let me remove those, as the fields will no longer be undefined or null.

Reply:
Does it really help in such cases? I have not used the `!` before but the StackOverflow sounded like it is just a TypeScript assertion (i.e. during static code checking not at runtime if something is undefined)?

Reply:
It did not help me. I still got the error 

I also used this first time by going through this - https://www.geeksforgeeks.org/typescript/typescript-non-null-assertion-operator-postfix-type/